java-checks/src/main/java/org/sonar/java/checks/MethodOnlyCallsSuperCheck.java

```diff
@@ -156,4 +161,14 @@ private static boolean hasAnnotationDifferentFromOverride(List<AnnotationTree> a
     return false;
   }
 
+  private static boolean isClassAnnotatedWithTransactional(MethodTree methodTree) {
+    Symbol.TypeSymbol typeSymbol = methodTree.symbol().enclosingClass();
+    if (typeSymbol != null){
+      SymbolMetadata metadata = typeSymbol.metadata();
+      return TRANSACTIONAL_ANNOTATIONS.stream().anyMatch(metadata::isAnnotatedWith);
+    }
+    // Defensive programming, should not be possible, enclosingClass returns null only for package symbol.
```

Comment:
I checked a couple of variants, this check seems to be redundant. Even if we put method outside of the class, we're not able to parse the file. So I can't imagine the situation where the enclosing class will be null. So let's remove this null check. If we see any problems on peach with it we still can add the check and at least find the proper test case.

Reply:
Okay, that makes sense. In addition, I double-checked and it seems there are other rules that do not perform the check, and nothing went wrong, so I believe it fine to not do it.